### PR TITLE
FlightLevel <-> Pressure MetPy conversion

### LINKED ISSUE #378: metpy
**[Original report](https://bitbucket.org/wxmetvis/mss/issue/378) by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

The metpy package looks veeery interstring to migrate a bit of our thermodynamics code and to have a proper unit conversion scheme in a single package.
Maintainer seems trustworthy, too!

### PR DESCRIPTION
**Pull request by Marilyth**
Convert flightlevel2pressure and pressure2flightlevel to metpy compatible unit functions for future PR.
Adjust MSS to work with it.
Sorry for the messy commit history, it turns out merging before a rebase is a lethal mistake.

### CODE REVIEW

File: mslib/utils.py [hunk 1]
```diff
@@ -32,6 +32,7 @@
 import logging
 import netCDF4 as nc
 import numpy as np
+from metpy.units import units
```
Comment (joernu76) [inline mslib/utils.py:35]: We need to switch the UR below to the registry of metpy.

File: mslib/_tests/test_thermolib.py [hunk 2]
```diff
@@ -27,69 +27,66 @@
...
 
 import mslib.thermolib as tl
 
 
 def test_flightlevel2pressure2flightlevel():
-    fs = np.arange(1, 71000, 1000.) / 30.48
-    ps = tl.flightlevel2pressure_a(fs)
-    fs_p = tl.pressure2flightlevel_a(ps)
-    assert fs == pytest.approx(fs_p)
+    fs = np.arange(1, 71000, 1000.) / 30.48 * units.hft
```
Comment (joernu76) [inline mslib/_tests/test_thermolib.py]: Probably better `(np.arange(...) * units.m).to(units.hft)`

File: mslib/msui/mpl_qtwidget.py [hunk 3]
```diff
@@ -576,31 +577,31 @@ def _determine_ticks_labels(self, typ):
...
             ma_dist, mi_dist = 4, 1.0
             if (top_km - bot_km) <= 20:
                 ma_dist, mi_dist = 1, 0.5
             elif (top_km - bot_km) <= 40:
                 ma_dist, mi_dist = 2, 0.5
             major_heights = np.arange(0, top_km + 1, ma_dist)
             minor_heights = np.arange(0, top_km + 1, mi_dist)
-            major_ticks = thermolib.flightlevel2pressure_a(major_heights / 0.03048)
-            minor_ticks = thermolib.flightlevel2pressure_a(minor_heights / 0.03048)
+            major_ticks = thermolib.flightlevel2pressure(major_heights * units.kilometer).magnitude
```
Comment (joernu76) [inline mslib/msui/mpl_qtwidget.py]: kilometer -> km
(consistency and brevity)

File: mslib/thermolib.py [hunk 4]
```diff
@@ -210,141 +211,126 @@ def omega_to_w(omega, p, t):
...
-    """
-    Conversion of flight level (given in hft) to pressure (Pa) with
-    hydrostatic equation, according to the profile of the ICAO
-    standard atmosphere.
-
-    Array version, the argument "flightlevel" must be a numpy array.
-
+# Values according to the 1976 U.S. Standard atmosphere [NOAA1976]_.
+# List of tuples (height, temperature, pressure, temperature gradient)
+standard_atmosphere = [
```
Comment (joernu76) [inline mslib/thermolib.py]: global variable, -> capital letters

File: mslib/thermolib.py [hunk 5]
```diff
@@ -210,141 +211,126 @@ def omega_to_w(omega, p, t):
...
 
     # Initialize the return array.
-    z = numpy.full_like(p, numpy.nan)
+    z = numpy.full_like(pressure, numpy.nan) * units.hft
 
-    for i, ((z0, t0, gamma, p0), (z1, t1, _, p1)) in enumerate(zip(ZTGPS[:-1], ZTGPS[1:])):
-        p1 = ZTGPS[i + 1][-1]
-        indices = (p > p1) & (p <= p0)
+    for i, ((z0, t0, p0, gamma), (z1, t1, p1, _)) in enumerate(zip(standard_atmosphere[:-1], standard_atmosphere[1:])):
+        p1 = standard_atmosphere[i + 1][-2]
```
Comment (joernu76) [inline mslib/thermolib.py]: maybe use named constants instead of -2?

**Review by joernu76 — CHANGES_REQUESTED** [review]

File: mslib/msui/mpl_qtwidget.py (hunk 3 above)
Comment (Marilyth) [reply mslib/msui/mpl_qtwidget.py]: Done

File: mslib/_tests/test_thermolib.py (hunk 2 above)
Comment (Marilyth) [reply mslib/_tests/test_thermolib.py]: Done

File: mslib/utils.py (hunk 1 above)
Comment (Marilyth) [reply mslib/utils.py:35]: Done

File: mslib/thermolib.py (hunk 4 above)
Comment (Marilyth) [reply mslib/thermolib.py]: Done

File: mslib/thermolib.py (hunk 5 above)
Comment (Marilyth) [reply mslib/thermolib.py]: Done

**Review by joernu76 — APPROVED** [review]

**Pull request merged**